packages/@aws-cdk/aws-ec2/lib/user-data.ts

```diff
@@ -51,14 +126,39 @@ export abstract class UserData {
    */
   public abstract addCommands(...commands: string[]): void;
 
+  /**
+   * Add one or more commands to the user data that will run when the script exits.
+   */
+  public abstract addOnExitCommands(...commands: string[]): void;
+
   /**
    * Render the UserData for use in a construct
    */
   public abstract render(): string;
+
+  /**
+   * Adds commands to download a file from S3
+   */
+  public abstract addS3DownloadCommand(params: S3DownloadOptions): string;
```

Comment:
It's not completely obvious what this returns, so that might be worth a `@returns` annotation.